atd-vze/src/views/Crashes/Maps/CrashEditCoordsMap.js

```diff
@@ -1,189 +1,130 @@
+  const handleFormCancel = () => {
+    setIsEditingCoords(false);
+  };
 
-    return (
-      <div>
-        <MapGL
-          {...viewport}
-          ref={this.mapRef}
-          width="100%"
-          height="350px"
-          mapStyle={
-            isDev
-              ? "mapbox://styles/mapbox/satellite-streets-v11"
-              : LOCATION_MAP_CONFIG.mapStyle
-          }
-          onViewportChange={this._updateViewport}
-          getCursor={this.getCursor}
-          // controller={customGeocoderMapController}
-          mapboxApiAccessToken={TOKEN}
-        >
-          {/* <Geocoder
-            mapRef={this.mapRef}
-            onViewportChange={this._handleViewportChange}
-            mapboxApiAccessToken={TOKEN}
-            inputValue={geocoderAddress}
-            options={{ flyTo: false }}
-            // Bounding box for auto-populated results in the search bar
-            bbox={[-98.22464, 29.959694, -97.226257, 30.687526]}
-          /> */}
-          <div className="fullscreen" style={fullscreenControlStyle}>
-            <FullscreenControl />
-          </div>
-          <div className="nav" style={navStyle}>
-            <NavigationControl showCompass={false} />
-          </div>
-          {/* add nearmap raster source and style */}
-          {!isDev && <LabeledAerialSourceAndLayer />}
-          <Marker latitude={markerLatitude} longitude={markerLongitude}>
-            <Pin size={40} isDragging={isDragging} animated />
-          </Marker>
-        </MapGL>
-        <CrashEditLatLonForm
-          latitude={markerLatitude}
-          longitude={markerLongitude}
-          handleFormSubmit={this.handleMapFormSubmit}
-          handleFormReset={this.handleMapFormReset}
-          handleFormCancel={this.handleMapFormCancel}
-        />
-      </div>
-    );
-  }
-}
+  return (
+    <>
+      <MapGL
+        ref={mapRef}
+        initialViewState={{
+          latitude: latitude_primary || defaultInitialState.latitude,
+          longitude: longitude_primary || defaultInitialState.longitude,
+          zoom: defaultInitialState.zoom,
+        }}
+        style={{ width: "100%", height: "50vh" }}
+        {...mapParameters}
+        cooperativeGestures={true}
+        draggable
+        onDragStart={() => setIsDragging(true)}
+        onDrag={onDrag}
+        onDragEnd={() => setIsDragging(false)}
+      >
+        <FullscreenControl position="top-left" />
+        <NavigationControl position="top-left" showCompass={false} />
+        <Marker {...markerCoordinates}>
+          <Pin size={40} color={"warning"} isDragging={isDragging} animated />
+        </Marker>
+        {/* add nearmap raster source and style */}
+        {!isDev && <LabeledAerialSourceAndLayer />}
+      </MapGL>
+      <CrashEditLatLonForm
+        {...markerCoordinates}
+        handleFormSubmit={handleFormSubmit}
+        handleFormReset={handleFormReset}
+        handleFormCancel={handleFormCancel}
+        isSubmitting={isSubmitting}
```

Comment:
I added this `isSubmitting` state so we can disable the form buttons while the new form values save. Open to any feedback here since it was out of scope.

Reply:
this is a really nice touch, great attention to detail!! :sparkles: